### PR TITLE
jquake: init at 1.6.1

### PR DESCRIPTION
**Pull request by nessdoor**
###### Motivation for this change
JQuake can be a very useful application for residents of Japan, and having it as a stand-alone application can be annoying due to the requirement for JRE 8. Its Nix expression turned out to be straightforward, and I don't see a reason for not having it here.

Creating a package that also works for Darwin should be easy, but I don't have a Darwin system, so help on this front is welcome.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/jquake/default.nix [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    sha256 = "0nw6xjc3i1b8rk15arc5d0ji2bycc40rz044qd03vzxvh0h8yvgl";
+  };
+
+  desktopFile = makeDesktopItem {
+    name = "JQuake";
+    desktopName = "JQuake";
+    exec = "JQuake";
+    comment = "Real-time earthquake map of Japan";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/jquake/default.nix:18]:
```suggestion
```

File: pkgs/applications/misc/jquake/default.nix [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    mv {JQuake.jar,JQuake_lib} $out/lib
+    mv sounds $out/lib
+
+    mkdir -p $out/share/licenses
+    mv LICENSE* $out/share/licenses
+
+    runHook postInstall
+  '';
+
+  desktopItems = [ desktopFile ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/jquake/default.nix:61]:
```suggestion
  desktopItems = [
    (makeDesktopItem {
      name = "JQuake";
      desktopName = "JQuake";
      exec = "JQuake";
      comment = "Real-time earthquake map of Japan";
    })
  ];
```

File: pkgs/applications/misc/jquake/default.nix [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    chmod +x JQuake.sh
+
+    mkdir -p $out/{bin,lib}
+    mv JQuake.sh $out/bin/JQuake
+    mv {JQuake.jar,JQuake_lib} $out/lib
+    mv sounds $out/lib
+
+    mkdir -p $out/share/licenses
+    mv LICENSE* $out/share/licenses
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/jquake/default.nix:56]: This probably collides with other packages.

File: pkgs/applications/misc/jquake/default.nix [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    # the appropriate level.
+    sed -i "/^java/ s/$/\ | logger -p user.debug/" JQuake.sh
+
+    substituteInPlace JQuake.sh \
+      --replace "java -jar " "exec ${jre8.outPath}/bin/java -jar $out/lib/"
+
+    # By default, an 'errors.log' file is created in the current directory.
+    # cd into a temporary directory and let it be created there.
+    substituteInPlace JQuake.sh \
+      --replace "[JAR FOLDER]" "\$(mktemp -p /tmp -d jquake-errlog-XXX)"
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/jquake/default.nix:37]:
```suggestion
    # By default, an 'errors.log' file is created in the current directory.
    # cd into a temporary directory and let it be created there.
    substituteInPlace JQuake.sh \
      --replace "java -jar " "exec ${jre8.outPath}/bin/java -jar $out/lib/" \
      --replace "[JAR FOLDER]" "\$(mktemp -p /tmp -d jquake-errlog-XXX)"
```

File: pkgs/applications/misc/jquake/default.nix [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    desktopName = "JQuake";
+    exec = "JQuake";
+    comment = "Real-time earthquake map of Japan";
+  };
+
+  nativeBuildInputs = [ unzip copyDesktopItems ];
+
+  sourceRoot = ".";
+
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/jquake/default.nix:23]:
```suggestion
  postPatch = ''
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/jquake/default.nix (hunk 3 above)
Comment (nessdoor) [reply pkgs/applications/misc/jquake/default.nix:56]: I didn't know where a more appropriate place could have been. Sorry, I never packaged something before.

Comment (SuperSandro2000) [reply pkgs/applications/misc/jquake/default.nix:56]:
```suggestion
    mkdir -p $out/share/licenses
    mv LICENSE* $out/share/licenses/jquake
```
yes please.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126842` at 268eb5f5 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>jquake</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md">patch-phase</a>

`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/applications/misc/jquake/default.nix:23:3:
```
   |
23 |   patchPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Pull request merged**